app/src/main/java/com/architectcoders/aacboard/datasource/local/LocationDataSourceImpl.kt

```diff
@@ -21,18 +22,22 @@ class LocationDataSourceImpl(private val context: Context) : LocationDataSource
     }
 
     @SuppressLint("MissingPermission")
-    override suspend fun getLastUserRegion(): String? =
+    override suspend fun getUserLanguage(): String? =
         suspendCancellableCoroutine { cancellableContinuation ->
             fusedLocationClient.lastLocation
                 .addOnCompleteListener {
-                    cancellableContinuation.resume(it.result.toRegion())
+                    cancellableContinuation.resume(it.result.toLanguage())
                 }
         }
 
-    private fun Location?.toRegion(): String? {
+    private fun Location?.toLanguage(): String? {
         val addresses = this?.let {
             geocoder.getFromLocation(latitude, longitude, 1)
         }
-        return addresses?.firstOrNull()?.countryCode
+        val countryCode = addresses?.firstOrNull()?.countryCode
+
+        return Locale.getAvailableLocales().firstOrNull { locale ->
+            locale.country == countryCode
+        }?.language
```

Comment:
If the user has set a language in the device that doesn't match his location we just go return a null language and there for we use `en`?

Maybe we could return the device default locale if there is no matching
```suggestion
        }?.language ?: Locale.getDefault().language
```

Reply:
Nice! Changed